Mathlib/Data/Set/Function.lean

```diff
@@ -1853,6 +1861,43 @@ protected lemma InvOn.extendDomain (h : InvOn g₁ g₂ s t) :
 
 end Set
 
+namespace Set
+variable {α₁ α₂ β₁ β₂ : Type*} {s₁ : Set α₁} {s₂ : Set α₂} {t₁ : Set β₁} {t₂ : Set β₂}
+  {f₁ : α₁ → β₁} {f₂ : α₂ → β₂} {g₁ : β₁ → α₁} {g₂ : β₂ → α₂}
+
+lemma InjOn.prodMap (h₁ : s₁.InjOn f₁) (h₂ : s₂.InjOn f₂) :
```

Comment:
I find dot notation for InjOn doesn't really read well, but won't block on it

Reply:
Why not use `Prod.map` in the statements?

Reply:
because it gets simplified to its definition as `Prod_map` (not a typo!!) is a simp lemma